language/vm/src/unit_tests/signature_token_tests.rs

```diff
@@ -14,29 +14,32 @@ fn serialize_and_deserialize_nested_types_max() {
     let mut ty = SignatureToken::Struct(StructHandleIndex::new(0));
     for _ in 1..SIGNATURE_TOKEN_DEPTH_MAX {
         ty = SignatureToken::Vector(Box::new(ty));
-    }
-
-    let mut binary = BinaryData::new();
-    serialize_signature_token(&mut binary, &ty).expect("serialization should succeed");
+        let mut binary = BinaryData::new();
+        serialize_signature_token(&mut binary, &ty).expect("serialization should succeed");
 
-    let mut cursor = Cursor::new(binary.as_inner());
-    load_signature_token(&mut cursor).expect("deserialization should succeed");
+        let mut cursor = Cursor::new(binary.as_inner());
+        load_signature_token(&mut cursor).expect("deserialization should succeed");
+    }
 }
 
 #[test]
 fn serialize_nested_types_too_deep() {
     let mut ty = SignatureToken::Struct(StructHandleIndex::new(0));
-    for _ in 0..SIGNATURE_TOKEN_DEPTH_MAX {
+    for _ in 1..SIGNATURE_TOKEN_DEPTH_MAX {
```

Comment:
Out of curiosity, is there also another test that makes sure that nested types are allowed up to this limit?

Reply:
no and I plan to add one, absolutely good point!
But overall I realized that coverage in deserializer is pretty high so I figured I'd spend the next few days looking for other places where we may not have a good enough coverage